File: farewell.js
var key = 'i hope in the next ten years there would be no other farewell letter brilliant than this one';

var fs = require('fs');
fs.readFile(process.argv[2], function (err, data) {
  var encrypted = data.toString().replace(/\s/g, '').split(',');
  var decrypted = '';
  for (var i = 0; i < encrypted.length; ++i) {
    decrypted += '%' + ('0' + (parseInt(encrypted[i]) ^ key.charCodeAt(i % key.length)).toString(16)).slice(-2);
  }
  process.stdout.write(decodeURIComponent(decrypted));
});

File: farewell.exs
defmodule Farewell do
  use Bitwise

  def say(file_name) do
    {:ok, content} = File.read file_name
    encrypted = Regex.split(~r/,/, Regex.replace(~r/\s/, content, ""))
    IO.write decrypt('i hope in the next ten years there would be no other farewell letter brilliant than this one',
                    0, encrypted, <<>>)
  end

  defp decrypt(_, _, [], decrypted), do: decrypted

  defp decrypt(key, index, [h|t], decrypted) do
    decrypt(key, index + 1, t,
            decrypted <>
            <<Enum.at(key, rem(index, length(key))) ^^^ String.to_integer(h)>>)
  end
end

Farewell.say Enum.at(System.argv, 0)